types/config.d.ts

```diff
@@ -42,16 +45,17 @@ export interface IConfig {
     // Send back instrumentation data, if set to true
     instrument?: boolean;
 
-    // Inspect CSSRuleList for style elements and send CSSRules data instead of style's text children
-    cssRules?: boolean;
-
     // Pointer to the function which would be responsible for sending the data
     // If left unspecified, raw payloads will be uploaded to the uploadUrl endpoint
     uploadHandler?: UploadHandler;
 
+    // XHLHttpRequest headers to be added to every upload request with the default upload handler
+    // Object is a map from header names to header values
+    uploadHeaders?: { [key: string]: string; };
+
     // Pointer to the function which will be responsible for giving Clarity
     // a dictionary of strings that the user wants logged in each Clarity payload
-    customInstrumentation?: () => { [key: string]: string; };
+    customInstrumentation?: (impressionId?: string, cid?: string, projectId?: string) => { [key: string]: string; };
```

Comment:
>customInstrumentation?: (impressionId?: string, cid?: string, projectId?: string) => { [key: string]: string; }; [](start = 4, length = 112)

I'm not sure I fully understand. Custom instrumentation is a webmaster-defined function that returns a dictionary. Since we call config.customInstrumentation() ourselves, we can *always* provide arguments above and it's up to webmasters whether to use them and return in a customInst dictionary or not, so these args should not be optional. 

You'd also need to update config.customInstrumentation() call in core.ts to pass these arguments.

Reply:
Agree with your first comment. GOes in line with Toby's comment. I will refactor. 

---
In reply to: [245132409](https://github.com/Microsoft/clarity-js/pull/116#discussion_r245132409) [](ancestors = 245132409)

Reply:
@Slava - The second part was already done.

---
In reply to: [245168785](https://github.com/Microsoft/clarity-js/pull/116#discussion_r245168785) [](ancestors = 245168785,245132409)

Reply:
Ah, right, i looked at the wrong line, sorry.

---
In reply to: [245169815](https://github.com/Microsoft/clarity-js/pull/116#discussion_r245169815) [](ancestors = 245169815,245168785,245132409)